src/runtime/runtime-debug.cc

```diff
@@ -972,9 +972,36 @@ static inline bool RecordReplayBytecodeAllowed() {
 
 extern bool gRecordReplayHasCheckpoint;
 
+// [RUN-1621] We use this to avoid warn spam.
+static bool hasWarnedUserJs = false;
+
 extern void RecordReplayOnTargetProgressReached();
 
 RUNTIME_FUNCTION(Runtime_RecordReplayAssertExecutionProgress) {
+  CHECK_EQ(1, args.length());
+  Handle<JSFunction> function = args.at<JSFunction>(0);
+  Handle<SharedFunctionInfo> shared(function->shared(), isolate);
+
+  if (recordreplay::AreEventsDisallowed() &&
+      !recordreplay::HasDivergedFromRecording() &&
+      function->shared().IsUserJavaScript() &&
+      function->shared().HasSourceCode()) {
```

Comment:
Unfortunately, Runtime_RecordReplayAssertExecutionProgress is very performance sensitive, and this will add too much overhead.  It is important that if additional testing flags like the progress assertions are not enabled, that this function only bump the progress counter and check against the target progress and then return (we want to inline this operation into jitcode but the attempts I've made to get that working have run into problems).

Only doing these checks when JS progress assertions are enabled would be fine, but then again in that case we'd get a warning anyways but without the stack information.

Reply:
I undid this.

This should be Ok. I already found several culprits that caused this and put them into RUN-1644. We can hopefully identify any other issues coming from this in our own tests, or come up with a more advanced strategy to very selectively enable this in customer executions if we hit a wall.